www/include/configuration/configKnowledge/display-services.php

```diff
@@ -113,83 +119,99 @@
+                LEFT JOIN host_service_relation hsr2 ON hsr2.service_service_id = s2.service_id
+                RIGHT JOIN hostgroup_relation hgr ON hgr.hostgroup_hg_id = hsr2.hostgroup_hg_id
+                LEFT JOIN host h2 ON h2.host_id = hgr.host_host_id
+            WHERE s2.service_register = '1' ";
     if ($postHostgroup !== false) {
-        $query .= "AND (h2.host_id IN ";
-        $query .= "(SELECT host_host_id FROM hostgroup_relation hgr
-        				 WHERE hgr.hostgroup_hg_id = :postHostgroup ) ";
-        $query .= "OR hgr.hostgroup_hg_id = :postHostgroup ) ";
+        $query .= "
+            AND (h2.host_id IN
+            (SELECT host_host_id FROM hostgroup_relation hgr
+                WHERE hgr.hostgroup_hg_id = :postHostgroup )
+            OR hgr.hostgroup_hg_id = :postHostgroup ) ";
     }
     if (!empty($postHost)) {
-        $query .= "AND h2.host_name LIKE  :postHost ";
+        $query .= "AND h2.host_name LIKE :postHost ";
     }
     if ($postServicegroup !== false) {
-        $query .= "AND s2.service_id IN ";
-        $query .= "(SELECT service_service_id FROM servicegroup_relation
-                     WHERE servicegroup_sg_id = :postServicegroup) ";
+        $query .= "
+            AND s2.service_id IN
+            (SELECT service_service_id FROM servicegroup_relation
+                WHERE servicegroup_sg_id = :postServicegroup) ";
     }
     if ($postPoller !== false) {
-        $query .= "AND h2.host_id IN ";
-        $query .= "(SELECT host_host_id FROM ns_host_relation
-        			WHERE nagios_server_id = :postPoller ) ";
+        $query .= "
+            AND h2.host_id IN
+            (SELECT host_host_id FROM ns_host_relation
+                WHERE nagios_server_id = :postPoller ) ";
     }
     if (!empty($postService)) {
         $query .= "AND s2.service_description LIKE :postService ";
     }
     $query .= ") as t1 ";
-    $query .= "ORDER BY $orderby " . $order . " LIMIT " . $num * $limit . ", " . $limit;
+    $query .= "ORDER BY $orderBy " . $order . " LIMIT :offset, :limit";
 
     $statement = $pearDB->prepare($query);
-    if (!empty($postHost)) {
-        $statement->bindValue(':postHost', $postHost, PDO::PARAM_STR);
-    }
-    if (!empty($postService)) {
-        $statement->bindValue(':postService', $postService, PDO::PARAM_STR);
-    }
-    if ($postPoller !== false) {
-        $statement->bindValue(':postPoller', $postPoller, PDO::PARAM_INT);
-    }
-    if ($postServicegroup !== false) {
-        $statement->bindValue(':postServicegroup', $postServicegroup, PDO::PARAM_INT);
-    }
-    if ($postHostgroup !== false) {
-        $statement->bindValue(':postHostgroup', $postHostgroup, PDO::PARAM_INT);
-    }
 
+    foreach ($queryValues as $bindId => $bindData) {
+        foreach ($bindData as $bindType => $bindValue) {
+            $statement->bindValue($bindId, $bindValue, $bindType);
+        }
+    }
+    $statement->bindValue(':offset', $num * $limit, PDO::PARAM_INT);
```

Comment:
just a comment : 
you could have used the same $queryValue array, before line 208 to add it to the foreach loop ;)